src/integrations/Http2Integration.ts

```diff
@@ -0,0 +1,291 @@
+import Integration from './Integration';
+import * as opentracing from 'opentracing';
+import {
+    Http2Tags,
+    SpanTags,
+    SpanTypes,
+    DomainNames,
+    ClassNames,
+    TriggerHeaderTags,
+} from '../Constants';
+import Utils from '../utils/Utils';
+import ModuleUtils from '../utils/ModuleUtils';
+import * as url from 'url';
+import ThundraLogger from '../ThundraLogger';
+import HttpError from '../error/HttpError';
+import ThundraSpan from '../opentracing/Span';
+import ThundraChaosError from '../error/ThundraChaosError';
+import ExecutionContextManager from '../context/ExecutionContextManager';
+
+const shimmer = require('shimmer');
+const has = require('lodash.has');
+const semver = require('semver');
+
+const thundraCollectorEndpointPattern1 = /^api[-\w]*\.thundra\.io$/;
+const thundraCollectorEndpointPattern2 = /^([\w-]+\.)?collector\.thundra\.io$/;
+
+const MODULE_NAME_HTTP2 = 'http2';
+
+class Http2Integration implements Integration {
+
+    config: any;
+    private instrumentContext: any;
+
+    constructor(config: any) {
+        ThundraLogger.debug('<HTTP2Integration> Activating HTTP2 integration');
+
+        this.config = config || {};
+        this.instrumentContext = ModuleUtils.instrument(
+            [MODULE_NAME_HTTP2], null,
+            (lib: any, cfg: any, moduleName: string) => {
+                this.wrap.call(this, lib, cfg, moduleName);
+            },
+            (lib: any, cfg: any, moduleName: string) => {
+                this.doUnwrap.call(this, lib, moduleName);
+            },
+            this.config);
+    }
+
+    static isValidUrl(host: string): boolean {
+        if (host.indexOf('amazonaws.com') !== -1) {
```

Comment:
Can we move `isValidUrl` logic to a common module (for example, `HTTPUtils`?) Because same logic is used by both HTTP and HTTP integrations

Reply:
method moved to new util HTTPUtils